src/Components/ModuleForm/ModuleForm.js

```diff
@@ -0,0 +1,310 @@
+    if (totalModules === 0) {
+      return 0;
+    }
+    const totalGrades = moduleList.reduce((acc, module) => {
+      switch (module.grade) {
+        case "A":
+          return acc + 5.0;
+        case "A-":
+          return acc + 4.5;
+        case "B+":
+          return acc + 4.0;
+        case "B":
+          return acc + 3.5;
+        case "B-":
+          return acc + 3.0;
+        case "C+":
+          return acc + 2.5;
+        case "C":
+          return acc + 2.0;
+        case "D":
+          return acc + 1.5;
+        default:
+          return acc;
+      }
+    }, 0);
+    return totalGrades / totalModules;
+  };
+
+  render() {
+    const { enabled, userAction, moduleName, moduleList, grade, error } =
+      this.state;
+    const gpa = this.calculateGPA();
+    return (
+      <div className="module-form-card">
+        <Card>
+          <form onSubmit={this.handleSubmit} className="module-form">
+            <input
+              type="text"
+              value={moduleName}
+              onChange={(e) => this.setState({ moduleName: e.target.value })}
+              className="module-name"
+              placeholder="Module Name"
+            />
+
+            <select
+              type="text"
+              value={grade}
+              onChange={(e) => this.setState({ grade: e.target.value })}
+              placeholder="Grade"
+              className="grade-selector"
+            >
+              Grade
+              <option value="A">A: 5.00</option>
+              <option value="A-">A-: 4.50</option>
+              <option value="B+">B+: 4.00</option>
+              <option value="B">B: 3.50</option>
+              <option value="B-">B-: 3.00</option>
+              <option value="C+">C+: 2.50</option>
+              <option value="C">C: 2.00</option>
+              <option value="D">D: 1.50</option>
+            </select>
+            <select
+              value={userAction}
+              onChange={this.handleActionChange}
+              className="action-selector"
+            >
+              <option value="Add">Add Module</option>
+              <option value="Update">Update Module</option>
+              <option value="Delete">Delete Module</option>
+            </select>
+            <Button
+              type="submit"
+              className={enabled > 0 ? "button-enabled" : "button-disabled"}
+              onClick={this.handleSubmit}
+            >
+              Let's go!
+            </Button>
+            {/* Display the module list */}
+          </form>
+        </Card>
+        {error ? <div className="error-message">{error}</div> : <div></div>}
+        <h2 className="module-list-title"> Module List</h2>
+        <Card>
+          <table>
+            <th className="module-list-name">
+              <td>
+                {" "}
+                <h3>Modules</h3>
+                <tr className="module-table-row-item">
+                  <ListGroup>
+                    {moduleList &&
```

Comment:
since moduleList defaults to empty array as per above definition `[]`, this statement will always be true

Reply:
Also, if you make the mapped list group item conditional, you should include the ListGroup and the tr tag above into the condition to avoid rendering those without any content